src/main/java/teammates/sqllogic/core/FeedbackSessionsLogic.java

```diff
@@ -386,4 +389,24 @@ public void adjustFeedbackSessionEmailStatusAfterUpdate(FeedbackSession session)
             session.setPublishedEmailSent(session.isPublished());
         }
     }
+
+    /**
+     * Gets a list of undeleted feedback sessions which start within the last 2 hours
+     * and need an open email to be sent.
+     */
+    public List<FeedbackSession> getFeedbackSessionsWhichNeedOpenEmailsToBeSent() {
+        List<FeedbackSession> sessionsToSendEmailsFor = new ArrayList<>();
+        List<FeedbackSession> sessions = fsDb.getFeedbackSessionsPossiblyNeedingOpenEmail();
+        log.info(String.format("Number of sessions under consideration: %d", sessions.size()));
+
+        for (FeedbackSession session : sessions) {
+            if (session.isOpened() && session.getCourse().getDeletedAt() == null) {
```

Comment:
Is it possible to check `isOpened` through Hibernate/SQL?

Reply:
@EuniceSim142 if you think it's too much change, we can leave this as a future issue, similar to [this](https://github.com/TEAMMATES/teammates/pull/12738#discussion_r1495257696) (maybe open an issue consolidating these so that we don't forget these potential optimisations - I suppose we'll only get to it after migration is merged)

But if you think it's fine we can do it in this PR as well - seems to just be transferring this logic to the db:
`(now.isAfter(startTime) || now.equals(startTime)) && now.isBefore(endTime);`

Personally prefer keeping it for a future PR (i.e. do an exact migration first, then optimise so that it's easier to ensure correctness), but up to you to make the decision

Reply:
I think similar to the discussion, i will leave this optimization for a future PR.